lms/djangoapps/certificates/tests/test_queue.py

```diff
@@ -110,18 +136,33 @@ def assert_queue_response(self, mode, expected_mode, expected_template_name):
                 mock_send.return_value = (0, None)
                 self.xqueue.add_cert(self.user_2, self.course.id)
 
-        # Verify that the task was sent to the queue with the correct callback URL
-        self.assertTrue(mock_send.called)
-        __, kwargs = mock_send.call_args_list[0]
-
-        actual_header = json.loads(kwargs['header'])
-        self.assertIn('https://edx.org/update_certificate?key=', actual_header['lms_callback_url'])
-        certificate = GeneratedCertificate.objects.get(user=self.user_2, course_id=self.course.id)
+        # Ensure that a certificate record has been created, and
+        # grading has been run. If the enrollment mode is eligible for
+        # a certificate, assert that it has been correctly
+        # generated. Otherwise, it should not generate a cert.
+        if CourseMode.is_eligible_for_certificate(mode):
+            # Verify that the task was sent to the queue with the correct callback URL
+            self.assertTrue(mock_send.called)
+
+            __, kwargs = mock_send.call_args_list[0]
+            actual_header = json.loads(kwargs['header'])
+            self.assertIn('https://edx.org/update_certificate?key=', actual_header['lms_callback_url'])
+            body = json.loads(kwargs['body'])
+            self.assertIn(expected_template_name, body['template_pdf'])
+
+            certificate = GeneratedCertificate.eligible_certificates.get(user=self.user_2, course_id=self.course.id)
+        else:
+            # Ensure the certificate was not generated
```

Comment:
So here we are no longer asserting the queue response, but that a certificate record is not present in the database -- seems like we're overloading this operation?